### PR TITLE
aten.squeeze.default and aten.squeeze.dim decomp

### PR DESCRIPTION
**Pull request by voznesenskym**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -13,6 +13,11 @@
 from torch._decomp import register_decomposition
 from torch._prims_common import NumberType, TensorLike, TensorSequenceType
 from torch._prims_common.wrappers import _maybe_resize_out, _safe_copy_out, out_wrapper
+from torch.fx.experimental.symbolic_shapes import (
```
Comment (voznesenskym) [inline torch/_decomp/decompositions.py:16]: woops, leftover from cherrypick

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1087,6 +1087,49 @@ def split_with_sizes(
...
+    strides = []
+
+    stride = self.stride()
+    breakpoint()
+    for i in range(0, n_dims):
+        size_at_i = size[i]
+        if size_at_i != 1:
+            sizes.append(size_at_i)
+            strides.append(stride[i])
+    breakpoint()
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:1106]: remove

File: torch/_decomp/decompositions.py [hunk 3]
```diff
@@ -1087,6 +1087,49 @@ def split_with_sizes(
...
+    return torch.as_strided(self, sizes, strides)
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:1107]: This is wrong when storage offset is nonzero

Comment (ezyang) [reply torch/_decomp/decompositions.py:1107]: actually it's fine

File: torch/_decomp/decompositions.py (hunk 2 above)
Comment (voznesenskym) [reply torch/_decomp/decompositions.py]: Done

File: torch/_decomp/decompositions.py [hunk 4]
```diff
@@ -1087,6 +1087,46 @@ def split_with_sizes(
...
+
+    stride = self.stride()
+
+    if size[dim] == 1:
+        l_size = list(size)
+        l_stride = list(stride)
+        del l_size[dim]
+        del l_stride[dim]
+        return torch.as_strided(self, l_size, l_stride)
+    return self
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:1127]: Does it return self or a fresh TensorImpl?

**Comment by ezyang** [comment]
tests?

File: torch/_decomp/decompositions.py (hunk 4 above)
Comment (gchanan) [reply torch/_decomp/decompositions.py:1127]: shouldn't all inplace methods return self?

File: torch/_decomp/decompositions.py [hunk 5]
```diff
@@ -1087,6 +1087,46 @@ def split_with_sizes(
...
+    return torch.as_strided(self, sizes, strides)
+
+
+@register_decomposition(aten.squeeze_.dim, disable_meta=True)
+def squeeze_dim(self, dim: int):
+    n_dims = self.dim()
+    size = self.size()
+    assert n_dims == len(size)
+    if n_dims == 0:
+        return self
```
[the remark this answers is not in the document]
Comment (voznesenskym) [reply torch/_decomp/decompositions.py:1113]: Do you mean the `dim` arg? Would the `dim` arg ever be valid / meaningful if n_dims is 0?

File: torch/_decomp/decompositions.py (hunk 4 above)
Comment (voznesenskym) [reply torch/_decomp/decompositions.py:1127]: self feels better here. Thoughts?

Comment (ezyang) [reply torch/_decomp/decompositions.py:1127]: This isn't an inplace method. For non-inplace method, view functions typically have the invariant that they always return a fresh TensorImpl (e.g., `resize_` the output doesn't affect input.)

**Pull request closed without merging**